src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -31,6 +35,161 @@ internal sealed class PdbWritingException : Exception
         }
     }
 
+    /// <summary>
+    /// A utility to log all operations and arguments to the native PDB writing
+    /// library, so that we can hash that log to generate a deterministic GUID and
+    /// timestamp.
+    /// </summary>
+    internal sealed class PdbLogger
+    {
+        // This class hashes the log data on-the-fly; see
+        // https://msdn.microsoft.com/en-us/library/system.security.cryptography.hashalgorithm.transformblock(v=vs.110).aspx
+        // That enables us to avoid producing a full log in memory.
+        private readonly bool _logging;
+        private readonly BinaryWriter _logData;
+        private const int bufferFlushLimit = 1024;
```

Comment:
Because of how slow it is to invoke methods through reflection, I'm considering increasing this substantially, like perhaps to 64k (64*1024).